src/Host.h

```diff
@@ -179,7 +179,12 @@ class Host : public QObject
     void registerAnonymousEventHandler(const QString& name, const QString& fun);
     void unregisterEventHandler(const QString&, TScript*);
     void raiseEvent(const TEvent& event);
-    void resetProfile();
+    // This disables all the triggers/timers/keys in preparation to resetting
+    // them:
+    void resetProfile_phase1();
+    // This actually does the bulk of the reset but must wait until the profile
+    // is quiescent:
+    void resetProfile_phase2();
```

Comment:
What does the waiting achieve? Could you add a comment?

Reply:
It is **not** safe to delete a script containing the `resetProfile()` command from the lua system - whilst it is ***executing***! So, *instead* of waiting to do the reset *until* a new text comes in and starts some triggers and stuff running we will now do it as soon as the processing of the `resetProfile()` and anything in the script that calls it has completed - which may be sooner, as it happens...

Reply:
Sounds good - can you put that in?

Reply:
Oh, I thought I had implied that - but I suppose not everyone realises that quiescent means "*not actively doing anything*"? :grinning:

One comment coming up...

Reply:
Done.